src/openmrs-esm-module-config.ts

```diff
@@ -0,0 +1,49 @@
+import * as R from "ramda";
+
+// The configurations that have been provided
+const configs : object[] = [];
+
+// An object with module names for keys and schemas for values.
+const schemas = {};
+
+export function defineConfigSchema(moduleName, schema) {
+  console.log("defineConfigSchema received schema for " + moduleName + ": " + JSON.stringify(schema));
+  schemas[moduleName] = schema;
+}
+
+export function provide(config) {
+  console.log("provide recieved config " + JSON.stringify(config));
+  configs.push(config);
+}
+
+export function getConfig(moduleName) {
+  if (!schemas.hasOwnProperty(moduleName)) {
+    throw Error("No config schema has been defined for " + moduleName);
+  }
+  const schema = schemas[moduleName];
+
+  // Create a config object composed of all the defaults
+  const defaultConfig = R.map(R.prop('default'), schema);
+
+  // Merge all of the configs provided for moduleName
+  const allConfigsForModule = R.map(R.prop(moduleName), configs)
+  const providedConfig = R.mergeAll(allConfigsForModule);
+
+  for (let [key, value] of Object.entries(providedConfig)) {
```

Comment:
This is not recursive and only checks top level properties. Are you planning on checking things deeper than top-level properties in a future PR? Or am I missing something about this code?

Reply:
Right, uh, I haven't figured out how we should support nesting in the configs. I'll do something in the brainstorming doc. I'll let you know when I've added an idea. Feel free to pitch in there, if you have ideas about how it should happen.

Reply:
https://issues.openmrs.org/browse/MF-87